spec/classes/dev_spec.rb

```diff
@@ -1,92 +1,57 @@
 require 'spec_helper'
 
 describe 'apache::dev', type: :class do
-  let(:pre_condition) do
-    [
-      'include apache',
-    ]
-  end
+  on_supported_os.each do |os, facts|
+    context "on #{os} " do
+      let :facts do
+        facts
+      end
 
-  context 'on a Debian OS' do
-    let :facts do
-      {
-        lsbdistcodename: 'jessie',
-        osfamily: 'Debian',
-        operatingsystem: 'Debian',
-        operatingsystemrelease: '6',
-        is_pe: false,
-        id: 'root',
-        path: '/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin',
-        kernel: 'Linux',
-      }
-    end
+      context 'with all defaults' do
+        let(:pre_condition) do
+          [
+            'include apache',
+          ]
+        end
 
-    it { is_expected.to contain_class('apache::params') }
-    it { is_expected.to contain_package('libaprutil1-dev') }
-    it { is_expected.to contain_package('libapr1-dev') }
-    it { is_expected.to contain_package('apache2-prefork-dev') }
-  end
-  context 'on an Ubuntu 16 OS' do
-    let :facts do
-      {
-        lsbdistrelease: '16.04',
-        lsbdistcodename: 'xenial',
-        osfamily: 'Debian',
-        operatingsystem: 'Ubuntu',
-        operatingsystemrelease: '16.04',
-        is_pe: false,
-        id: 'root',
-        path: '/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin',
-        kernel: 'Linux',
-      }
+        case facts[:os]['name']
+        when 'Debian'
+          context 'on a Debian OS' do
```

Comment:
We already know this context so it's redundant (see line 4). I won't repeat this for every spec, but it's a repeated pattern.

Reply:
Removed the redundant context's